apex/contrib/optimizers/distributed_fused_adam.py

```diff
@@ -786,7 +1058,7 @@ def _init_param_state(
                     main_param_fragment = bucket.params_shard[shard_start:shard_end]
                     main_param_fragment.copy_(model_param_fragment)
 
-    def zero_grad(self, set_to_none=True):
+    def zero_grad(self, set_to_none=False):
```

Comment:
is this change to make use of the contiguous buffer for params & grads?

Reply:
Yep, `set_to_none=False` avoids an extra memory allocation and memory copy if `contiguous_grad_buffer=True` and should have no impact if `contiguous_grad_buffer=False`. It also matches [`torch.optim.Adam`](https://pytorch.org/docs/stable/generated/torch.optim.Adam.html#torch.optim.Adam.zero_grad).